ui/src/components/VoxettaRecordButton.js

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+import {LitElement, html, css} from 'lit-element';
+import {styleMap} from 'lit-html/directives/style-map.js';
+
+import {AudioRecorder} from '../utils/AudioRecorder';
+import {UtteranceApiService} from '../utils/UtteranceApiService';
+
+import {Icon} from '@material/mwc-icon';
+
+// Styling for the button when the user is not recording
+let nonRecordingStyle = { 
+    backgroundColor: 'white', 
+    color: '#3c4043' 
+};
+
+// Styling for the button when the user is recording
+let recordingStyle = { 
+    backgroundColor: 'red', 
+    color: 'white' 
+};
+
+/**
+ * Button responsible for enabling the user to record and upload audio files. 
+ */
+export class VoxettaRecordButton extends LitElement {
+    static get properties() {
+        return {
+            isRecording: {type: Boolean},
+            audioStream: {type: Object}
+        };
+    }
+
+    static get styles() {
+        return css`
+            div {
+                align-items: center; 
+                display: flex; 
+                flex-direction: column; 
+                flex-wrap: wrap;
+                justify-content: center; 
+                text-align: center;  
+            }
+            mwc-icon-button {
+                --mdc-icon-button-size: 80px;
+                --mdc-icon-size: 40px; 
+                border-radius: 50%;
+                box-shadow: 0px 8px 15px #dcdcdc;
+                color: #3c4043;
+            }
+        `;
+    }
+
+    constructor() {
+        super();
+        this.isRecording = false;
+        this.audioRecorder = new AudioRecorder();
+        this.utteranceService = new UtteranceApiService();
+        this.audioStream; 
+    }
+
+    updated() {
+        this.handleWaveCanvas();
+    }
+  
+    /**
+     * If the user is not currently recording, begin recording using the Microphone 
+     * component. Otherwise, stop recording and save and display the just-recorded 
+     * audio file.
+     */
+    async recordHandler() {
+        if (!this.isRecording) {
+            try { 
+                await this.audioRecorder.initRecorder();
+            } catch(e) { 
+                console.log('Microphone blocked.'); 
+                return;
+            }
+
+            if (!this.audioRecorder.startRecording()) {
+                 console.log('Failed to start recording.')
+                 return; 
+            }
```

Comment:
is this where we plan to alert the user if their microphone is blocked? if so, will this be changed to reflect that? or will I be doing that in styling?

Reply:
I think the old alert messages got lost in the merge conflicts--changed from console.logs to alerts, but def change to whatever you think looks/flows best when styling